packages/compiler/src/core/checker.ts

```diff
@@ -3582,7 +3585,7 @@ export function createChecker(program: Program): Checker {
       return links.declaredType as UnionVariant;
     }
 
-    const name = variantNode.id.sv;
+    const name = variantNode.id ? variantNode.id.sv : Symbol("name");
```

Comment:
Does this mean that anonymous variants will all now have a symbol called `name` representing their key in the union?

Reply:
Indeed it does. Is that bad?

Reply:
Sorry to be clear, all anonymous variants of union declarations. Anonymous variants of union expressions do not change.